### PR TITLE
Feat/synthetic print delegate stx

### LINKED ISSUE #3465: [Stacks 2.1] Lack of synthetic event attached to Delegate Stacks STX Bitcoin op
It looks like the stacks-node is not attaching a synthetic event when crafting synthetic transactions created for representing Bitcoin operations performing a delegate Stacks STX.
This synthetic events are essentials for downstream event observers: transactions can't be classified without these, a typical stacks-api node would block its observed stacks-node, resulting in frozen stacks explorers, etc.  
cc @zone117x for confirmation.

### PR DESCRIPTION
**Pull request by pavitthrap**
### Description
This PR adds a synthetic print event to the `delegate-stx` function in the `pox-2` contract. It does so by means of the special contract handler (defined in `special.rs`). 

This PR adds 2 tests:
- a test in `pox_2_tests.rs` that called `delegate-stx` through a normal stacks transaction
- a test in `neon_integrations.rs` that sends a btc wire DelegateStx tx, and verifies (a) that the delegation was successful, and (b) that a synthetic print event was emitted. 

### Applicable issues
- fixes #3465

### Checklist
- [x] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [x] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: src/clarity_vm/special.rs [hunk 1]
```diff
@@ -566,6 +593,7 @@ fn synthesize_pox_2_event_info(
         "stack-aggregation-commit"
         | "stack-aggregation-commit-indexed"
         | "stack-aggregation-increase" => Some(create_event_info_aggregation_code(function_name)),
+        "delegate-stx" => Some(create_event_info_stack_or_delegate_code(sender, function_name, args)),
```
Comment (zone117x) [inline src/clarity_vm/special.rs:596]: Should this be moved to the first match condition which already calls `create_event_info_stack_or_delegate_code`?

**Comment by pavitthrap** [comment]
> Hey @pavitthrap I'm trying to test out this branch and running into this build error:
> 
> ```
> error[E0599]: no method named `build_delegate_stacks_tx` found for mutable reference `&mut BitcoinRegtestController` in the current scope
>     --> testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:1771:22
>      |
> 1771 |                 self.build_delegate_stacks_tx(epoch_id, payload, op_signer, None)
>      |                      ^^^^^^^^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `build_pre_stacks_tx`
> ```

Just fixed the non-test build!

**Review by jcnelson — APPROVED** [review]
This LGTM as long as @zone117x and @lgalabru are able to confirm that the event observer receives the requisite data.  Thanks for tackling this!

**Review by zone117x — APPROVED** [review]
Tested and working as expected at https://github.com/hirosystems/stacks-blockchain-api/pull/1527

Thanks @pavitthrap!

**Pull request merged**